### ISSUE TITLE
The base !help-command does not work in staff channels

### ISSUE THREAD

**Issue by SebastiaanZ** [python-discord/bot#262]
At the moment, the base `!help` command does not work in staff channels. A cursory search in staff indicates that it broke somewhere after 2018-12-17, as this is the last time it was successful there:

![2019-01-09_17-52-27](https://user-images.githubusercontent.com/33516116/50914799-ac6b9800-1437-11e9-8b2a-05b5dac45f87.png)

Currently, the bot just doesn't respond when `!help` is called without arguments. Strangely enough, it does work when used with an argument, .e.g, `!help unban`, whether called directly or as fallback when a command is run without the proper arguments, e.g. `!unban` without arguments.

I've not found any obvious reason for it, but I know @scragly also looked, so maybe he can share his findings as well.

**Comment by SebastiaanZ** [comment]
Okay, the base command can only be called from bot-commands at the moment, so the problem is not restricted to the staff channels. Seems like a whitelisting problem somewhere.

**Comment by SebastiaanZ** [comment]
Running the command `!help` on my test/sandbox server show this in the debug logs:

```
Jan 09 18:29:47 pd.beardfist.com Bot: |                 bot.decorators |    DEBUG | Ves Zappa#3787 tried to call the 'charinfo' command. The command was not used in a whitelisted channel, but the author had a role to bypass the in_channel check.
```

That's weird. I'll look into it.

**Comment by SebastiaanZ** [comment]
Okay, I think I know what's going on.

The help command checks which commands a user can run to determine which should be in the help embed. However, the way it does that throws the `bot.decorators.InChannelCheckFailure` exception for some commands when executed outside of bot-commands. Since that exception is not handled, the help command stops executing.

I'll work on a patch.

**Comment by sco1** [comment]
Closed by #263
